core/config/project_settings.cpp

```diff
@@ -68,12 +68,27 @@ String ProjectSettings::get_resource_path() const {
 	return resource_path;
 }
 
-String ProjectSettings::get_safe_project_name() const {
-	String safe_name = OS::get_singleton()->get_safe_dir_name(get("application/config/name"));
-	if (safe_name.is_empty()) {
-		safe_name = "UnnamedProject";
+String ProjectSettings::get_csharp_project_name() const {
```

Comment:
I don't know if the Core team will be happy with a C# related method in Project Settings. I think the `get_safe_project_name` method was meant to be agnostic since we are outside the .NET module.

And it seems to me that the only reason why this method is here is for the purpose of detecting C# usage for the project features. Maybe modules should be able to manipulate project features to customize/extend them.

Reply:
How about moving this method into the .NET module? Then ProjectSettings can check for the C# feature by looking for any .csproj file inside the project directory, instead of checking the exact path.

```diff
diff --git a/core/config/project_settings.cpp b/core/config/project_settings.cpp
index cb25dc9ebf..7e9d7dd98e 100644
--- a/core/config/project_settings.cpp
+++ b/core/config/project_settings.cpp
@@ -934,6 +934,23 @@ Error ProjectSettings::_save_custom_bnd(const String &p_file) { // add other par
 	return save_custom(p_file);
 }
 
+#ifdef TOOLS_ENABLED
+bool _csproj_exists(String p_root_dir) {
+	Ref<DirAccess> dir = DirAccess::open(p_root_dir);
+
+	dir->list_dir_begin();
+	String file_name = dir->_get_next();
+	while (file_name != "") {
+		if (!dir->current_is_dir() && file_name.get_extension() == "csproj") {
+			return true;
+		}
+		file_name = dir->_get_next();
+	}
+
+	return false;
+}
+#endif // TOOLS_ENABLED
+
 Error ProjectSettings::save_custom(const String &p_path, const CustomMap &p_custom, const Vector<String> &p_custom_features, bool p_merge_with_current) {
 	ERR_FAIL_COND_V_MSG(p_path.is_empty(), ERR_INVALID_PARAMETER, "Project settings save path cannot be empty.");
 
@@ -952,7 +969,7 @@ Error ProjectSettings::save_custom(const String &p_path, const CustomMap &p_cust
 		}
 	}
 	// Check for the existence of a csproj file.
-	if (FileAccess::exists(get_resource_path().path_join(get_safe_project_name() + ".csproj"))) {
+	if (_csproj_exists(get_resource_path())) {
 		// If there is a csproj file, add the C# feature if it doesn't already exist.
 		if (!project_features.has("C#")) {
 			project_features.append("C#");

```